docroot/modules/humsci/hs_migrate/hs_migrate.module

```diff
@@ -87,14 +88,36 @@ function hs_migrate_entity_delete(EntityInterface $entity) {
 }
 
 /**
- * Implements hook_cron().
+ * Ultimate cron callback function to execute a migration group.
+ *
+ * The entity id should be in the form `hs_migrate_{migrate_group_id}` to
+ * execute that group.
+ *
+ * @param \Drupal\ultimate_cron\CronJobInterface $cron_entity
+ *   Ultimate cron entity.
  */
-function hs_migrate_cron() {
+function hs_migrate_ultimate_cron_task(CronJobInterface $cron_entity) {
+  $logger = \Drupal::logger('hs_migrate');
   $migrations = hs_migrate_migration_list();
-  // Take it one group at a time, importing the migrations within each group.
-  foreach ($migrations as $migration_list) {
-    array_walk($migration_list, 'hs_migrate_execute_migration');
+
+  $migration_group = str_replace('hs_migrate_', '', $cron_entity->id());
+  // Execute the migration entities in the provided migration group.
+  if ($migration_group && !empty($migrations[$migration_group])) {
+    try {
+      array_walk($migrations[$migration_group], 'hs_migrate_execute_migration');
```

Comment:
Does this create blocking iteration? As in, if you have 4 migrations to run and the second one throws an error does this block the third and fourth execution? You may actually want to refactor this to use a loop or send in an anonymous function with the try/catch in it.

Reply:
it would only block migrations within the same group. for example theres a `hs_capx_images` importer and a `hs_capx`. if the images fail to import, the profiles wont import. which in my mind is an ideal thing

Reply:
That totally makes sense. Thanks. I'll run through the test steps now.